### ISSUE TITLE
add a new (spoken) language

### ISSUE THREAD

**Issue by goatshriek** [goatshriek/stumpless#117]
If you know a language other than English and you're looking for a way to contribute to a project without knowing much about coding, adding a new locale is a great place to start.

There is a detailed guide on how to go about adding a new language that you know in the [localization documentation](https://github.com/goatshriek/stumpless/blob/latest/docs/localization.md). If you're feeling up to it, go ahead and open a new issue with the language you would like to add, and go to it!

Is the language that you wanted already added? Look over the locale header for it anyway (in `include/private/config/locale/`); any strings added after it was initially added will still be English placeholders marked with a `// todo translate` comment. A contribution translating these is almost as important as the initial translation, and is a much simpler change. Rather than implementing a completely new set of headers, you will just change the placeholder strings to be locale-specific strings instead. If you want to see a list of what strings need translating by locale, you can also use grep to get a current list:

```sh
grep -r -c "// todo translate" include/private/config/locale/

# output looks like this:
# include/private/config/locale/bg-bg.h:28
# include/private/config/locale/cz-cz.h:15
# include/private/config/locale/de-de.h:36
# include/private/config/locale/el-gr.h:16
# include/private/config/locale/en-us.h:0
# include/private/config/locale/es-es.h:18
# include/private/config/locale/fr-fr.h:36
# include/private/config/locale/it-it.h:0
# include/private/config/locale/pl-pl.h:17
# include/private/config/locale/sk-sk.h:31
# include/private/config/locale/sv-se.h:36
```

If there aren't any for your language or you'd rather do something new, have a look in the `l10n` folder of the repository. This has localized documentation for the project, and there are lots of opportunities to add a paragraph or two of translation. Don't feel like you need to translate an entire file for this, just a few paragraphs is more than enough! The `l10n/zh-cn` folder is a good example of what this could look like as more translations are added.

**Comment by Lockna** [comment]
I would like to add a German translation ^^

**Comment by goatshriek** [comment]
@Lockna sounds good! I have created #121 for the specific locale. I also added a reference to a recent commit that has examples of a similar update recently, if you want a clean example to complete the documentation. Any suggestions on improvements to the documentation/process are also welcome!

**Comment by pepo939393** [comment]
I would like to add Slovak translation

**Comment by goatshriek** [comment]
@pepo939393 Awesome! I just created #137 for you, it has a reference to the most recent locale addition as an example.

**Comment by taliezincho** [comment]
I'll try to add Bulgarian :)

best regards

**Comment by goatshriek** [comment]
Sounds good! I created #174 for you with some references and an example commit.

**Comment by palike608** [comment]
Hi,  i would like to add a Czech translation. Can i?

**Comment by goatshriek** [comment]
Absolutely! I will create an issue for it in the next 4 hours or so.

**Comment by goatshriek** [comment]
I've created #190 to track the Czech translation.

**Comment by gcotofernandez** [comment]
Hi! I would like to add a Spanish translation

**Comment by goatshriek** [comment]
Hi Gabriela! Thanks so much for reaching out! A spanish translation does exist already, but there are a few strings that have been added since then that need attention: check out `include/private/config/locale/es-es.h` to see what's there. I suspect you could make quick work of it!

If you are looking for something more substantial there will be an internationalization framework for the documentation soon, but it isn't in place just yet. I can accelerate that if it is something you are interested in - it would be on the scale of picking a function or two (or however much you want!) and providing a translation for that documentation.

**Comment by Marecek460** [comment]
I would like to add Polish translation?

**Comment by goatshriek** [comment]
Absolutely! I've created #199 to track this.

**Comment by SwoleDuck** [comment]
Hello, could i add a greek translation?

**Comment by goatshriek** [comment]
Hello @SwoleDuck, absolutely! I've created #240 to track that.

**Comment by lhardt** [comment]
Hello! I would like to add the Portuguese-Brazil translations!

Could you add the tags for hacktoberfest on this?

**Comment by goatshriek** [comment]
Absolutely @lhardt! I've created #275 to track this - once the PR is submitted and merged I'll add the tags to that so that you get credit.

**Comment by Santhusha-bit** [comment]
Hello! I would like to add the Sinhala(Sri Lanka) translations!
Could you add the tags for hacktoberfest on this?

**Comment by goatshriek** [comment]
Sure thing, @Santhusha-bit! I've created #283 to track this, and will add the tags to the PR once it is merged for hacktoberfest.

**Comment by viveknshah** [comment]
Hi @goatshriek , I have followed your guidance to create #284 and solve via #285 . Let me know if that works or any feedback since this is my first time in this space.

**Comment by goatshriek** [comment]
Awesome @viveknshah, thanks! I've left a review on the PR, once those changes are made you should be good to go.

**Comment by das-amit** [comment]
Hey @goatshriek , I've added Bengali Translations.
Have a look #291

**Comment by m20027** [comment]
Japanese is not yet?

**Comment by goatshriek** [comment]
Hello @m20027, Japanese has not been added yet! I've created #301 if you would like to tackle this.

**Comment by idpolik** [comment]
I'd like to add Hungarian :)

**Comment by goatshriek** [comment]
Sure thing! I've added #315 to track that.

**Comment by goatshriek** [comment]
Sounds good! I created #316 to track Swahili, but have left out the language tag for you to fill in depending on the specifics of the locale, as the [IANA registry](https://www.iana.org/assignments/language-subtag-registry/language-subtag-registry) has many different options for it.

**Comment by ilallou** [comment]
Hello can I contribute to this project by adding albanian translation?

**Comment by goatshriek** [comment]
Absolutely! I've created #324 to track the Albanian translation.

**Comment by rishabhtomar800** [comment]
I would love to add Hindi.

**Comment by goatshriek** [comment]
There is an existing Hindi translation at [include/private/config/locale/hi-in.h](https://github.com/goatshriek/stumpless/blob/latest/include/private/config/locale/hi-in.h), but it is missing a few translations. A pull request to translate the missing strings would be appreciated!

**Comment by LasseEmilBuch** [comment]
Hi!

i wouldnt mind doing some translation to danish :)

**Comment by goatshriek** [comment]
Hello @LasseBuchAspIT that would be great! I have created #343 to track it.

**Comment by neversaynv3r** [comment]
hi! I would like to finish the spanish localization, I've seen there are still some strings left

**Comment by AssafBardugo** [comment]
Hi, I would like to add a Hebrew translation

**Comment by goatshriek** [comment]
Sounds good! I've added #350 to track the Hebrew translation.

**Comment by OsmanBuyuksar** [comment]
Hi, i would like to add Turkish translation

**Comment by goatshriek** [comment]
Sure thing, I've created #388 to track the Turkish localization.

**Comment by AryasCodeTreks** [comment]
Hello @goatshriek 
Have added Telugu Localization
Please have a look https://github.com/goatshriek/stumpless/pull/407

**Comment by goatshriek** [comment]
Reopening (closed by accident).

**Comment by pmj-chosim** [comment]
Hello:) I want to add Korean translation

**Comment by goatshriek** [comment]
Hello @pmj-chosim, thanks for reaching out!

There is an existing Korean translation used for localization at `include/private/config/locale/ko-kr.h`. There is one string that needs to be translated currently, and your contribution for that would be welcome!

If you'd like to do more than that one string, please have a look at the `l10n/ko-kr`. This is the localized documentation for the project, and is right now mostly empty. If you would like, you can rename the README file there and translate the key features and "what can it log to" sections. You can refer to `l10n/zh-cn` for an example of what this would look like.

**Comment by pmj-chosim** [comment]
Yes, thank you! I compared the en-us and ko-kr locale files (include/private/config/locale/~), but it was hard to find the specific string that needs to be translated. Could you give me some advice? 
And I'm currently working on the |10n/ko-kr job(specificly: |10n/ko-kr/README.md). My translation isn't complete yet, but can I ask some questions about localization through the PR window?

**Comment by goatshriek** [comment]
To find strings that need to be translated you can do a search for "todo" in the ko-kr.h file. The string will have the English text instead of the Korean, so you would just change it directly in this file and be good to go.

Yes, feel free to open a PR and ask any questions there! Don't feel like you have to do the ENTIRE readme (it is quite long) but of course we will appreciate however much you can do!

**Comment by GiovannaSLisboa** [comment]
Olá! Gostaria de realizar as contribuições para a tradução em português-BR. Identifiquei que ainda não está completa

**Comment by MR-martiny** [comment]
Hello! I Would also love to help with de english-portuguese translation!

**Comment by goatshriek** [comment]
My apologies for the delay @GiovannaSLisboa and @MR-martiny, please feel free to submit a pull request with any translation updates!

`<via translate.google.com>`
Peço desculpas pela demora, @GiovannaSLisboa e @MR-martiny. Sintam-se à vontade para enviar um pull request com quaisquer atualizações de tradução!

**Comment by c0mplex10** [comment]
I want to add Turkish but I don't know where to start

**Comment by goatshriek** [comment]
There is a Turkish translation already, but there are some missing strings that would be helpful to add. You would simply to update the strings marked with a "todo translate" comment in the [Turkish header](https://github.com/goatshriek/stumpless/blob/latest/include/private/config/locale/tr-tr.h), as described in the localization documentation mentioned in the primary issue description.

**Comment by mklinzing** [comment]
I would like to add Irish Gaelic

**Comment by goatshriek** [comment]
Hi @mklinzing, please feel free to add Irish Gaelic; I believe the language tag would be ga-ie.

**Comment by goatshriek** [comment]
Accidental closure.

**Comment by ebazomik** [comment]
Hi @goatshriek, I'd like to contribute to the Italian language.

**Comment by begineri** [comment]
Hi @goatshriek, I'd like to contribute to the Chinese language.

**Comment by DeltaPhi0** [comment]
Hello sir! Is Italian still available? I could start working on it if I'm not stepping on anyone's foot for it!

**Comment by goatshriek** [comment]
Sure, please feel free to submit anything for Italian! There are a few outstanding strings in the `it-it.h` locale header, or some localization can be done to add an `it-it` folder in the `l10n` folder with a translation of the main README.

**Comment by ebazomik** [comment]
> Hello sir! Is Italian still available? I could start working on it if I'm not stepping on anyone's foot for it!

@DeltaPhi0 I saw that you've already created the pull request for the Italian language. I was waiting for a response too, but it looks like I was too slow ;)

**Comment by goatshriek** [comment]
@ebazomik my apologies for the delay in response, I often need a few days to find time to respond.

Please don't feel like there is no Italian translation work left! One thing that would be super helpful is to add a localization of the `stumpless.h` header. You can use `include/stumpless.h` as the reference, and put the translation into `l10n/it-it/<include>/stumpless.h`. An example of this in a different language is `l10n/zh-cn/包含/stumpless.h`.

**Comment by ebazomik** [comment]
Thanks @goatshriek  for the suggestion, I'll get started on it.

**Comment by DanC271** [comment]
I was going to ask about Italian, but I see they recently started offering it. Thanks anyway! 👍

**Comment by Lin-ZJ** [comment]
Hi! I’d like to work on a Simplified Chinese localization PR for this issue.

Since the zh-CN locale header already appears translated, I plan to translate `docs/CONTRIBUTING.md` into `l10n/zh-cn/文档/贡献指南.md` and update the related links in the zh-CN README.

Does that scope sound OK?
